modules/postgres-upgrade.nix

```diff
@@ -0,0 +1,56 @@
+# This module exposes a script to ease postgres upgrades.
+# It is based off the NixOS manual: https://nixos.org/manual/nixos/stable/#module-services-postgres-upgrading
+
+{ config
+, lib
+, options
+, pkgs
+, ...
+}:
+let
+  cfg = config.services.postgresql;
+  oldPostgres = cfg.finalPackage;
```

Comment:
I expected this to use `package` not `finalPackage`, based on what's in `common/services.nix`.